File: Non-overlapping-Intervals.java
1class Solution {
2    public int eraseOverlapIntervals(int[][] intervals) {
3       Arrays.sort(intervals, (a, b) -> Integer.compare(a[1], b[1]));
4
5       int removed = 0 ;
6       int last = intervals[0][1];
7
8       for(int i = 1;i<intervals.length ;i++){
9        if(intervals[i][0]>= last){
10            last = intervals[i][1];
11        }else{
12            removed++; 
13        }
14       }
15        return removed;
16    }
17}

File: Non-overlapping-Intervals.cpp
1class Solution {
2public:
3    bool static myComparator(const vector<int>& a, const vector<int>& b) {
4        return a[0] < b[0];
5    }
6    int eraseOverlapIntervals(vector<vector<int>>& intervals) {
7
8        sort(intervals.begin(), intervals.end(), myComparator);
9        int count = 0 ; 
10        int pre = intervals[0][1];
11        for(int i = 1;i<intervals.size();i++){
12            if(intervals[i][0]>=pre){
13                pre = intervals[i][1];
14            }else{
15                count++;  
16                pre = min(pre, intervals[i][1]);
17            }
18        }
19        return count; 
20    }
21};